### ISSUE TITLE
[bug] Conan v2 think that macOS SDK is an iOS SDK

### ISSUE THREAD

**Issue by paulocoutinhox** [conan-io/conan#12260]
### Environment Details (include every applicable attribute)
  * Operating System+version: macos latest
  * Compiler+version: clang 14
  * Conan version: 1.53
  * Python version: 3.9

### Steps to reproduce (Include if Applicable)

```
git clone https://github.com/nativium/nativium.git -b fix-conan-1-53
cd nativium
python3 nativium.py conan setup
python3 nativium.py target ios setup --arch=arm64 --build=debug --group=ios
python3 nativium.py target ios build --arch=arm64 --build=debug --group=ios
```

### Logs (Executed commands with output) (Include/Attach if Applicable)

```
CMake Error at /opt/homebrew/Cellar/cmake/3.24.2/share/cmake/Modules/Platform/iOS-Initialize.cmake:4 (message):
  
  /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk
  is not an iOS SDK
Call Stack (most recent call first):
  /opt/homebrew/Cellar/cmake/3.24.2/share/cmake/Modules/CMakeSystemSpecificInitialize.cmake:21 (include)
  CMakeLists.txt:8 (project)
```

The problem is here:
https://github.com/nativium/nativium/blob/fix-conan-1-53/conan/darwin-toolchain/conanfile.py#L102

The `XCRun` object from conan don't check correctly the settings. It get the settings from conanfile, but need be something like i do, that is try to get `settings_target`. Or make XCRun receive `settings` and not `conanfile`:

```
settings_target = None

if hasattr(self, "settings_target"):
    settings_target = self.settings_target
    self.output.info("Settings: Target")
else:
    settings_target = self.settings
    self.output.info("Settings: Default")
```

The conan XCRun lines is here:
https://github.com/conan-io/conan/blob/develop/conan/tools/apple/apple.py#L82-L87

**Comment by paulocoutinhox** [comment]
Hi,

Any news about this?

Thanks.

**Comment by czoido** [comment]
Hi @paulocoutinhox,

Yes, we have assigned this issue to the 1.54 milestone so hopefully we will have a fix for the next release. Sorry for the inconvenience and thanks a lot.

**Comment by czoido** [comment]
Hi @paulocoutinhox, I have just opened a PR form 1.54 solving this issue: https://github.com/conan-io/conan/pull/12319 Could you please have a look? Thanks!

**Comment by paulocoutinhox** [comment]
Hi @czoido. I tested here and is ok.

If anyone want test, install `conan` from this command:

```
python3 -m pip install git+https://github.com/czoido/conan@fix/12260       
```

**Comment by czoido** [comment]
Hi @paulocoutinhox,
We have merged the feature in 1.54 that will be released very soon. Please note that we finally changed the default to use `use_settings_target=False`. Thanks a lot for reporting and testing this.
